synapse/storage/databases/main/push_rule.py

```diff
@@ -50,60 +62,31 @@
 logger = logging.getLogger(__name__)
 
 
-def _is_experimental_rule_enabled(
-    rule_id: str, experimental_config: ExperimentalConfig
-) -> bool:
-    """Used by `_load_rules` to filter out experimental rules when they
-    have not been enabled.
-    """
-    if (
-        rule_id == "global/override/.org.matrix.msc3786.rule.room.server_acl"
-        and not experimental_config.msc3786_enabled
-    ):
-        return False
-    if (
-        rule_id == "global/underride/.org.matrix.msc3772.thread_reply"
-        and not experimental_config.msc3772_enabled
-    ):
-        return False
-    return True
-
-
 def _load_rules(
     rawrules: List[JsonDict],
     enabled_map: Dict[str, bool],
     experimental_config: ExperimentalConfig,
-) -> List[JsonDict]:
+) -> FilteredPushRules:
+    """Take the DB rows returned from the DB and convert them into a full
+    `FilteredPushRules` object.
+    """
+
     ruleslist = []
     for rawrule in rawrules:
-        rule = dict(rawrule)
-        rule["conditions"] = db_to_json(rawrule["conditions"])
-        rule["actions"] = db_to_json(rawrule["actions"])
-        rule["default"] = False
-        ruleslist.append(rule)
-
-    # We're going to be mutating this a lot, so copy it. We also filter out
-    # any experimental default push rules that aren't enabled.
-    rules = [
-        rule
-        for rule in list_with_base_rules(ruleslist)
-        if _is_experimental_rule_enabled(rule["rule_id"], experimental_config)
-    ]
-
-    for i, rule in enumerate(rules):
-        rule_id = rule["rule_id"]
+        ruleslist.append(
+            PushRule(
+                rule_id=rawrule["rule_id"],
+                priority_class=rawrule["priority_class"],
+                conditions=db_to_json(rawrule["conditions"]),
+                actions=db_to_json(rawrule["actions"]),
+            )
+        )
```

Comment:
I think you could do this as a comprehension if you want now, theoretically that's slightly faster?

```suggestion
    ruleslist = [
        PushRule(
            rule_id=rawrule["rule_id"],
            priority_class=rawrule["priority_class"],
            conditions=db_to_json(rawrule["conditions"]),
            actions=db_to_json(rawrule["actions"]),
        ) for rawrule in rawrules
     ]
```